Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -93,23 +93,20 @@ import Foundation
     /// cached on subsequent calls for better performance.
     @_documentation(visibility: private)
     public func fetchOrReturnRemoteConfiguration(_ completion: @escaping (BTConfiguration?, Error?) -> Void) {
-        configurationLoader.getConfig { [weak self] configuration, error in
-            guard let self else {
-                completion(nil, BTAPIClientError.deallocated)
-                return
-            }
-            
-            if let error {
+        // TODO: - Consider updating all feature clients to use async version of this method?
+
+        Task { @MainActor in
+            do {
+                let configuration = try await configurationLoader.getConfig()
+                setupHTTPCredentials(configuration)
+                completion(configuration, nil)
+            } catch {
                 completion(nil, error)
-                return
             }
-            
-            setupHTTPCredentials(configuration)
-            completion(configuration, nil)
         }
     }
     
-    func fetchConfiguration() async throws -> BTConfiguration {
+    @MainActor func fetchConfiguration() async throws -> BTConfiguration {
```

Comment:
im a still thinking about this. that it might block main thread. 
Can we try to move @MainActor from the top Task block and just run the completion and setupHTTPCredentials on main?